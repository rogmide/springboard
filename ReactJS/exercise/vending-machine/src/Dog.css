.img-holder-dog {
  display: inline-grid;
  margin-top: 80px;
  width: 70vw;
  height: 70vh;
  border-radius: 10px;
  background: url(https://images.unsplash.com/photo-1552053831-71594a27632d?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=662&q=80)
    no-repeat center center fixed;
  -webkit-background-size: cover;
  -moz-background-size: cover;
  -o-background-size: cover;
  background-size: cover;
  justify-items: center;
  justify-content: space-around;
  align-content: center;
  align-items: start;
}

h1 {
  padding: 10px;
  font-weight: bolder;
  color: #bdd3d3;
  box-shadow: rgba(0, 0, 0, 0.4) 0px 3px 8px;
}

a {
  padding: 5px;
  font-size: 22px;
  font-weight: bolder;
  color: #bdd3d3;
  box-shadow: rgba(15, 2, 2, 0.9) 0px 3px 8px;
}
