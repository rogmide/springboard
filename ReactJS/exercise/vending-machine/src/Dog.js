import React from "react";
import { Link } from "react-router-dom";
import "./Dog.css";

const Dog = () => {
  return (
    <div>
      <div className="img-holder-dog">
        <br></br>
        <h1>Dogs Are So Nice</h1>
        <p>Not availableÂ in a vendor machine sorry.</p>
        <Link exact="true" to="/">
          Home
        </Link>
      </div>
    </div>
  );
};

export default Dog;
