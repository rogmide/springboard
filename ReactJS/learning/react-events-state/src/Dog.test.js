import React from "react";
import { render, screen } from "@testing-library/react";
import Dog from "./Dog";

test("should render", () => {
  render(<Dog name="Rusty" isAdopted={true} />);
});

test("should match snapshot", () => {
  const { asFragment } = render(<Dog name="Rusty" isAdopted={true} />);
  expect(asFragment()).toMatchSnapshot();
});

test("should match snapshot", () => {
  const { asFragment } = render(<Dog name="Rusty" isAdopted={false} />);
  expect(asFragment()).toMatchSnapshot();
});
