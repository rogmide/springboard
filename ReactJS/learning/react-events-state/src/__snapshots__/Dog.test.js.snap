// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Dog"
  >
    <h1>
      Rusty
    </h1>
    <p>
      has been adopted!
    </p>
  </div>
</DocumentFragment>
`;

exports[`should match snapshot 2`] = `
<DocumentFragment>
  <div
    class="Dog"
  >
    <h1>
      Rusty
    </h1>
    <button>
      Adopt me!
    </button>
  </div>
</DocumentFragment>
`;
